src/main/java/org/opensearch/geospatial/index/query/xypoint/XYPointQueryVisitor.java

```diff
@@ -0,0 +1,196 @@
+    @Override
+    public Query visit(MultiLine multiLine) {
+        return geometryNotSupported(ShapeType.MULTILINESTRING);
+    }
+
+    /**
+     * @param multiPoint  input geometry {@link MultiPoint}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     * throws QueryShardException
+     */
+    @Override
+    public Query visit(MultiPoint multiPoint) {
+        return geometryNotSupported(ShapeType.MULTIPOINT);
+    }
+
+    /**
+     * @param point  input geometry {@link Point}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     * throws QueryShardException
+     */
+    @Override
+    public Query visit(Point point) {
+        return geometryNotSupported(ShapeType.POINT);
+    }
+
+    /**
+     * @param circle  input geometry {@link Circle}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     * throws QueryShardException
+     */
+    @Override
+    public Query visit(Circle circle) throws RuntimeException {
+        return geometryNotSupported(ShapeType.CIRCLE);
+    }
+
+    /**
+     * @param rectangle  input geometry {@link Rectangle}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     */
+    @Override
+    public Query visit(Rectangle rectangle) {
+        Objects.requireNonNull(rectangle, "Rectangle cannot be null");
+        XYRectangle xyRectangle = toXYRectangle(rectangle);
+        var query = XYPointField.newBoxQuery(fieldName, xyRectangle.minX, xyRectangle.maxX, xyRectangle.minY, xyRectangle.maxY);
+        if (!fieldType.hasDocValues()) {
+            return query;
+        }
+
+        var dvQuery = XYDocValuesField.newSlowBoxQuery(fieldName, xyRectangle.minX, xyRectangle.maxX, xyRectangle.minY, xyRectangle.maxY);
+        return new IndexOrDocValuesQuery(query, dvQuery);
+    }
+
+    /**
+     * @param multiPolygon  input geometry {@link MultiPolygon}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     */
+    @Override
+    public Query visit(MultiPolygon multiPolygon) {
+        Objects.requireNonNull(multiPolygon, "Multi Polygon cannot be null");
+        return visitCollection(multiPolygon);
+    }
+
+    /**
+     * @param polygon  input geometry {@link Polygon}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     */
+    @Override
+    public Query visit(Polygon polygon) {
+        Objects.requireNonNull(polygon, "Polygon cannot be null");
+        return visitCollection(new GeometryCollection(Collections.singletonList(polygon)));
+    }
+
+    /**
+     * @param collection  input geometry {@link GeometryCollection}
+     * @return {@link Query} instance from XYPointField.XYPointInGeometryQuery
+     */
+    @Override
+    public Query visit(GeometryCollection<?> collection) {
+        if (collection.isEmpty()) {
+            return new MatchNoDocsQuery();
+        }
+        var bqb = new BooleanQuery.Builder();
```

Comment:
let's give valid name to variables
```suggestion
        var queryBuilder = new BooleanQuery.Builder();
```

Reply:
updated